js/react.md

```diff
@@ -102,6 +102,111 @@ The signUpFields array in turn consist of an array of objects, each describing a
 
 {% include sign-up-fields.html %}
 
+The following example will replace all the default sign up fields with the ones defined in the `signUpFields` array. It will also indicate that the `Email` field will be used to sign up with.
+```js
+import React, { Component } from 'react';
+import { withAuthenticator } from 'aws-amplify-react';
+
+class App extends Component {
+}
+
+const signUpConfig = {
+  header: 'My Customized Sign Up',
+  hideAllDefaults: true,
+  defaultCountryCode: '1',
+  signUpFields: [
+    {
+      label: 'Email',
+      key: 'email',
+      required: true,
+      displayOrder: 1,
+      type: 'string',
+      signUpWith: true
+    },
+    {
+      label: 'Password',
+      key: 'password',
+      required: true,
+      displayOrder: 2,
+      type: 'password'
+    },
+    {
+      label: 'PhoneNumber',
+      key: 'phone_number',
+      required: true,
+      displayOrder: 3,
+      type: 'string'
+    },
+    {
+      label: 'Custom Attribute',
+      key: 'custom_attr',
+      required: false,
+      displayOrder: 4,
+      type: 'string',
+      custom: true
+    }
+  ]
+};
+
+export default withAuthenticator(App, { signUpConfig });
+```
+
+#### Sign up/in with email/phone number
+If the user pool is set to allow email addresses/phone numbers as the username, you can then change the UI components accordingly by using `usernameAttributes`.
+
+When you are using `email` as the username:
+```js
+import { withAuthenticator, Authenticator } from 'aws-amplfy-react';
+// import { withAuthenticator, Authenticator } from 'aws-amplify-react-native';
```

Comment:
I think we should split up the react and react-native examples. Doing it all in one example is confusing to the customer

Reply:
Also we should leave out commented code.